src/_igraph/graphobject.c

```diff
@@ -13748,6 +13748,149 @@ PyObject *igraphmodule_Graph_community_leiden(igraphmodule_GraphObject *self,
   return error ? NULL : Py_BuildValue("Nd", res, (double) quality);
 }
 
+/**
+ * Voronoi clustering
+ */
+PyObject *igraphmodule_Graph_community_voronoi(igraphmodule_GraphObject *self,
+                                               PyObject *args, PyObject *kwds) {
+    static char *kwlist[] = {"modularity", "lengths", "weights", "mode", "radius", NULL};
+    PyObject *lengths_o = Py_None, *weights_o = Py_None;
+    PyObject *mode_o = Py_None;
+    PyObject *radius_o = Py_None;
+    PyObject *modularity_o = Py_None;
+    igraph_vector_t *lengths_v = NULL;
+    igraph_vector_t *weights_v = NULL;
+    igraph_vector_int_t membership_v, generators_v;
+    igraph_neimode_t mode = IGRAPH_ALL;
+    igraph_real_t radius = -1.0;  /* negative means auto-optimize */
+    igraph_real_t modularity = IGRAPH_NAN;
+    PyObject *membership_o, *generators_o, *result_o;
+    igraph_bool_t return_modularity = false;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|OOOOO", kwlist,
+                                     &modularity_o, &lengths_o, &weights_o, &mode_o, &radius_o))
+        return NULL;
+
+    if (modularity_o != Py_None){
+      modularity = (igraph_real_t)PyFloat_AsDouble(modularity_o);
+    }
+    else {
+      return_modularity = true;
+    }
+
+    /* Handle mode parameter */
+    if (mode_o != Py_None) {
+        if (igraphmodule_PyObject_to_neimode_t(mode_o, &mode))
```

Comment:
No need to check for `Py_None` when calling our conversion functions that convert a `PyObject` to an enum. All our conversion functions that convert to an enum check whether the input is `Py_None`, and if so, _they do not touch the current value_ in the enum output argument. So, it is enough to set up `mode` such that it has the default value that you want to assign to `Py_None` and then you can safely call the conversion function wihtout a check for `Py_None`.